models/DataObject/ClassDefinition/CustomLayout.php

```diff
@@ -118,6 +118,35 @@ public static function getById($id)
         return $customLayout;
     }
 
+    /**
+     * @param $name
+     *
+     * @throws \Exception
+     * @return null|CustomLayout|mixed
+     */
+    public static function getByName($name)
+    {
+        if ($name === null) {
+            throw new \Exception('Custom layout name is null');
+        }
+
+        try {
+            $customLayout = new self();
+            $customLayout->getDao()->getByName($name);
+
+            if ($customLayout->getId()) {
+                DataObject\Service::synchronizeCustomLayout($customLayout);
+                \Pimcore\Cache\Runtime::set($customLayout->getId(), $customLayout);
```

Comment:
this is dangerous because `id` is not unique systemwide, isn't it? 

in general, I would suggest that we implement the getByName method in a similar way as e.g. in ClassDefinition here https://github.com/pimcore/pimcore/blob/5d373e1085dc0f5e17306d5301a3ed90f63e9d00/models/DataObject/ClassDefinition.php#L224 ...
just get the id for a name and then call getById(). so we would avoid code duplication and also inconsistencies like the cache id.

Reply:
I will check it tomorrow. Sorry for the late reply.....pretty busy right now due of high workload in my job.

Reply:
no worries.

Reply:
Done